### ISSUE TITLE
Event's indexed arguments cause decodeEvent to fail if is an array type

### ISSUE THREAD

**Issue by hayeah** [trufflesuite/truffle-contract#45]
If indexed event argument is string or bytes (dynamic array), the topic is the the Keccak-256 hash, not the original value.

> If arrays (including string and bytes) are used as indexed arguments, the Keccak-256 hash of it is stored as topic instead. ([see doc](https://solidity.readthedocs.io/en/develop/contracts.html#events))

Given this event:

```
event EventUserPublicKeyUpdate(string indexed name, bytes32 key);
```

Would cause this error:

```
1) Contract: TreeholeUsers should register public key with username:
     Uncaught Error: Number can only safely store up to 53 bits
      at assert (node_modules/ethjs-abi/node_modules/bn.js/lib/bn.js:6:21)
      at BN.toNumber (node_modules/ethjs-abi/node_modules/bn.js/lib/bn.js:506:7)
      at node_modules/ethjs-abi/lib/index.js:82:59
      at Array.forEach (native)
      at decodeParams (node_modules/ethjs-abi/lib/index.js:78:9)
      at Object.decodeEvent (node_modules/ethjs-abi/lib/index.js:123:10)
      at node_modules/truffle-contract/contract.js:73:38
      at Array.map (native)
      at Object.decodeLogs (node_modules/truffle-contract/contract.js:44:19)
      at Object.callback (node_modules/truffle-contract/contract.js:172:35)
```

The root cause is [ethpm-js](https://github.com/trufflesuite/ethpm-js) using type "string" from the ABI to decode a bytes32 value.

**Comment by pszabop** [comment]
I might have a related issue:   indexed arguments are printed as Javascript integers, and not a bignum, which means I actually lose bits if my indexed argument is truly 256 bits.

for example logging `_id` using the Solidity event:

 `event LogValues(uint256 indexed _id, bytes32[] _values);`

and value:
```
`0xff2b5a83953910cb3600b2dc37ec9b9964b2e671e379260c99b9385d45ed6e6d`
```
results in:
```
"_id": "1.15416375517157630993985776368731714810906104678660497488521569475675349151341e+77"
```

**Comment by pszabop** [comment]
It looks like it's because you are using a very old package for the ABI decoder: 
`"ethjs-abi": "0.1.8",`

I'm having a hard time tracking which version it's fixed in but there was a lot of work in Februrary in this package around indexed log items 0.1.8 isn't even listed as a release or branch


[ethjs-abi on github](https://github.com/ethjs/ethjs-abi)

**Comment by cgewecke** [comment]
@hayeah Hi - thanks for opening. This issue is being tracked at [truffle 728](https://github.com/trufflesuite/truffle/issues/728). 

TLDR; is that `web3` cannot manage indexing for dynamic datatypes like strings and arrays. Only fixed size types can be indexed.
